internal/topology/variables/cluster_variable_defaulting_test.go

```diff
@@ -188,6 +188,38 @@ func Test_DefaultClusterVariables(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "Don't default variables set to null",
+			clusterClassVariables: []clusterv1.ClusterClassVariable{
+				{
+					Name:     "cpu",
+					Required: true,
+					Schema: clusterv1.VariableSchema{
+						OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+							Type:     "integer",
+							Default:  &apiextensionsv1.JSON{Raw: []byte(`1`)},
+							Nullable: true,
+						},
+					},
+				},
+			},
+			clusterVariables: []clusterv1.ClusterVariable{
+				{
+					Name: "cpu",
+					Value: apiextensionsv1.JSON{
+						Raw: nil,
+					},
+				},
+			},
+			want: []clusterv1.ClusterVariable{
+				{
+					Name: "cpu",
+					Value: apiextensionsv1.JSON{
+						Raw: nil,
+					},
```

Comment:
I was expecting the default to be applied... am I missing something?

Reply:
The default is applied it the variable is not set. But the variable is set to nil (the value is mandatory and if the value `null` is set via YAML this will result in a nil array)

Reply:
I added a comment about what the nil array means.